src/edu/ucsb/nceas/metacat/util/SystemUtil.java

```diff
@@ -125,55 +125,36 @@ public static String discoverServerSSLPort(HttpServletRequest request) {
     }
 
     /**
-     * Get the server URL which is made up of the server name + : + the http
-     * port number. Note that if the port is 80, it is left off.
+     * Get the server URL which is made up of the http or https + :// + the server name + : + the 
+     * port number. Note that if the port is 80 or 443, it is left off.
      * 
      * @return string holding the server URL
      */
     public static String getServerURL() throws PropertyNotFoundException {
-        String httpPort = PropertyService.getProperty("server.httpPort");
-        
-        String serverURL = "http://";
-        if(httpPort.equals("443") || httpPort.equals("8443"))
-        {
-            serverURL = "https://";
-        }
-        
-        serverURL += PropertyService.getProperty("server.name");
-        
-        if (!httpPort.equals("80") && !httpPort.equals("443")) {
-            serverURL += ":" + httpPort;
+        String serverURL = "https://";
+        String serverHttps = PropertyService.getProperty("server.https");
+        if (serverHttps != null && serverHttps.equalsIgnoreCase("false")) {
```

Comment:
Thank you for the information! This is my bad habit to check for `null` before comparing strings :). I think this is a defensive way - it wouldn't hurt anything but if the behavior of `PropertyService.getProperty()` changes, our code wouldn't throw a null exception. What do you think?

Reply:
> This is my bad habit to check for null before comparing strings :).

Don't get me wrong - I think this is a very good habit. However, in cases where an API makes a guarantee (e.g. [getProperty() will never return null](https://github.com/NCEAS/metacat/blob/1f770e5dbb8dcd8b028289c3dc9af10402b6a4a9/src/edu/ucsb/nceas/metacat/properties/PropertyService.java#L164)), then I think it is better to remove the null check. That makes the code simpler and easier to read, and it also maintains consistency (i.e. most places in the code do not have a null check when calling `getProperty()`). Otherwise, someone new to the code wonders why there's a null check (when the method never returns null), and this leads to cognitive dissonance, confusion (feelings of "did I miss something important??") and therefore wasted time).

I hear your concern about the possibility of getProperty() starting to return `null` in future, but that is extremely unlikely. If it does change, it will be a huge refactor anyway 😬. 

In summary, I think it's generally better to have consistent, readable, understandable code, instead of adding contingencies for a possible future change that is very unlikely to happen.

Reply:
Good points all around. I also note that wrapping the null check up with the empty string check could both make it more readable when we need to do both, and standardize the check. Apache common-lang even has some `StringUtils` methods for this common need, although they are so simple it would likely be better to just implement them in our software directly.

`StringUtils.isEmpty(String str)` - Checks if a String is null or empty ("")
`StringUtils.isBlank(String str)` - Checks if a String is null, whitespace (like " "), or empty ("")